contracts/TsteemPool.sol

```diff
@@ -0,0 +1,289 @@
+pragma solidity ^0.5.0;
+
+import "./SafeMath.sol";
+import "./Ownable.sol";
+
+interface ERC20Token {
+    function transfer(address _to, uint256 _value) external returns (bool);
+
+    function transferFrom(
+        address sender,
+        address recipient,
+        uint256 amount
+    ) external returns (bool);
+
+    function balanceOf(address account) external view returns (uint256);
+}
+
+interface PeanutsPool {
+    function withdrawPeanuts() external;
+
+    function getPendingPeanuts() external view returns (uint256);
+}
+
+contract TsteemPool is Ownable {
+    using SafeMath for uint256;
+    struct Delegator {
+        uint256 tsteemAmount; // deposted tsteem
+        uint256 availablePeanuts; // minted peanuts for user
+        uint256 debtRewards; // rewards debt
+        bool hasDeposited; // set true when first time deposit
+    }
+    mapping(address => Delegator) public delegators;
+    address[] public delegatorsList;
+    uint256 public shareAcc;
+    uint256 public totalDepositedTSTEEM;
+    address public devAddress;
+    uint256 public lastRewardBlock;
+
+    ERC20Token Pnuts;
+    ERC20Token Tsteem;
+    PeanutsPool PnutPool;
+
+    event DepositTSTEEM(address delegator, uint256 amount);
+    event WithdrawTSTEEM(address delegator, uint256 amount);
+    event WithdrawPeanuts(address delegator, uint256 amount);
+
+    modifier onlyDelegator() {
+        require(
+            delegators[msg.sender].hasDeposited,
+            "Account is not a delegator"
+        );
+        _;
+    }
+
+    constructor(
+        address _punts,
+        address _pnutPool,
+        address _tsteem
+    ) public {
+        Pnuts = ERC20Token(_punts);
+        Tsteem = ERC20Token(_tsteem);
+        PnutPool = PeanutsPool(_pnutPool);
+
+        shareAcc = 0;
+        totalDepositedTSTEEM = 0;
+        lastRewardBlock = 0;
+    }
+
+    // should do this before game start
+    function setDevAddress(address dev) public onlyOwner {
```

Comment:
Maybe we should set a dev address in constructor in advance. Case there is a period that after we deployed contract and before we call setDevAddress, user can deposit or withdraw TSTEEM, but devAddress is 0. Another solution is to add initialize function, all stuff should be prevented

Reply:
Yes, i agree with you.